transaction/transaction.go

```diff
@@ -20,33 +20,39 @@ const (
 //
 // Nodes is a union of the all the TxnStep.Nodes
 type Txn struct {
+	// TODO: Any good reason for this to be not just string ?
 	ID    uuid.UUID
 	Ctx   TxnCtx
 	Steps []*Step
 	Nodes []uuid.UUID
 }
 
 // NewTxn returns an initialized Txn without any steps
-func NewTxn() *Txn {
+func NewTxn(id string) *Txn {
 	t := new(Txn)
-	t.ID = uuid.NewRandom()
-	prefix := txnPrefix + t.ID.String()
 
+	parsedID := uuid.Parse(id)
+	if parsedID != nil {
```

Comment:
You could simplify this. `t.ID` has a nil zero value. So you could just parse directly into it.
```
if t.ID = uuid.Parse(); t.ID == nil {
  t.ID = uuid.NewRandom()
}